### PR TITLE
chore: temporarily disable circle filter to build & push PRs

### PR DESCRIPTION
**Pull request by lukebond**
We will be deploying a canary image to one ingester only in one environment and rather than build the image locally this PR will enable image building and pushing for PRs temporarily so we can get an image for a branch that we don't want to merge ([this one](https://github.com/influxdata/influxdb_iox/pull/5958)).

I'll revert it once we have that image built.

UPDATE: i changed this on request to be parameterised and not temporary. in fact that's what `build_perf` already did, so i renamed that.

- [ ] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

**Comment by crepererum** [comment]
Could we have a parameter-based, non-temporary solution? I think we had this a while back which allowed team members to issue a circleCi REST call to build and push images. Let me check where this is gone...

**Comment by crepererum** [comment]
Ah, it's still there:

https://github.com/influxdata/influxdb_iox/blob/99f36f88897c5a3f1fce05881cf891ce4a7f9191/.circleci/config.yml#L512-L523

**Comment by lukebond** [comment]
> Ah, it's still there:

i think it's for the CI base image. but i'll do something similar for this build and deploy step :+1:

**Review by domodwyer — APPROVED** [review]
Sounds legit - thanks for this :+1:

File: .circleci/config.yml [hunk 1]
```diff
@@ -509,6 +521,13 @@ workflows:
     jobs:
       - ci_image
 
+  # Force build and push of container image for non-main branch
+  release_branch:
```
Comment (crepererum) [inline .circleci/config.yml:525]: Isn't this what `build_perf` below does? Also `deploy_release` requires `build_release` to work.

**Review by crepererum — CHANGES_REQUESTED** [review]

**Review by crepererum — APPROVED** [review]
Thanks :)

**Pull request merged**